core/src/main/java/com/klaviyo/coresdk/networking/requests/KlaviyoRequest.kt

```diff
@@ -1,15 +1,22 @@
 package com.klaviyo.coresdk.networking.requests
 
+import com.klaviyo.coresdk.ConfigFileUtils
 import com.klaviyo.coresdk.networking.NetworkBatcher
 
 internal abstract class KlaviyoRequest: NetworkRequest() {
     companion object {
         internal const val BASE_URL = "https://a.klaviyo.com"
+
+        internal const val ANON_KEY = "\$anonymous"
     }
 
     override var queryData: String? = null
     override var payload: String? = null
 
+    internal fun addAnonymousIdToProps(map: MutableMap<String, String>) {
+        map[ANON_KEY] = "Android:${ConfigFileUtils.readOrCreateUUID()}"
```

Comment:
QUESTION: Is there a way to namespace the context this is created within? My concern is that for apps that want to allow users to login/logout that they would appear as the same anonymous ID until we discover their phone number or email. Its not critical to address now, but we should probably at least consider it as an option now.

Reply:
So, when I RFC'd my proposed solution for this issue there was some contentious aspects to it that we couldn't resolve there and then. What we decided to do in the meantime was mimic what the iOS SDK does for anonymous IDs for the early stages (which is what brings us to this drawback with logins/logouts) and to revisit the anonymous IDs later

Reply:
And just to keep you in the loop on the details, my proposed solution was to use lifecycle listeners to generate new anonymous IDs for each "user session" (I defined user sessions as ending after 30-60 seconds of being away from the parent application).
The worry was the number of anonymous profiles I could end up generating in Klaviyo if users had applications that don't collect much user info.

Reply:
Ah thats right, I forgot about that. Carry on!